From: Steve Langasek <steve.langasek@canonical.com>
Date: Sat, 7 Sep 2013 00:00:00 +0200
Subject: Fix calculation of the size of our signature data

The 'size' field of the certificate table header includes the size of
the header itself.  When parsing a signed file, we should therefore
subtract the size of this header from the field representing the size
of the pkcs7 data packet; otherwise when we detach (and subsequently
reattach) a signature, we wind up with 8 extra bytes of zeroes at the
end each time.  Fixing this ensures that detaching and signature and
then reattaching it to the file gives us back the original file.
---
 src/image.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/image.c b/src/image.c
index f3d4baa..a34f117 100644
--- a/src/image.c
+++ b/src/image.c
@@ -285,7 +285,7 @@ static int image_pecoff_parse(struct image *image)
 	if (cert_table && cert_table->revision == CERT_TABLE_REVISION &&
 			cert_table->type == CERT_TABLE_TYPE_PKCS &&
 			cert_table->size < size) {
-		image->sigsize = cert_table->size;
+		image->sigsize = cert_table->size - sizeof(*cert_table);
 		image->sigbuf = talloc_memdup(image, cert_table + 1,
 				image->sigsize);
 	}
